### PR TITLE
[adsp] fix startup crash and startup add-on checks, improve loops

### PR DESCRIPTION
**Pull request by AlwinEsch**
This changes fix a crash of kodi if dsp system is enabled.

The `CApplicationMessenger::GetInstance().SendMsg(TMSG_SETAUDIODSPSTATE, ACTIVE_AE_DSP_STATE_ON);` was called two times on kodi's start and on one place it becomes removed in this request.

Also is a check added to prevent a activate/deactivate if already in this state.

Further is the place with wrong labels reported on #7562 removed, is not needed and was to report dsp enabled with no add-ons installed and to inform system becomes disabled.

On the last patch I'm unsure what is better, is about the add-on folder name which becomes selected if nothing is enabled.
What is better?
"addons://user/kodi.adsp" (Which is currently used from me)
or
"addons://" (To have everything)

This need also be fixed on PVR system and after I know the best folder name comes a other request for there.

### CODE REVIEW

File: xbmc/cores/AudioEngine/DSPAddons/ActiveAEDSP.cpp [hunk 1]
```diff
@@ -93,16 +92,19 @@ void CActiveAEDSP::OnApplicationMessage(KODI::MESSAGING::ThreadMessage* pMsg)
     case TMSG_SETAUDIODSPSTATE:
       if(pMsg->param1 == ACTIVE_AE_DSP_STATE_ON)
       {
-        if(pMsg->param1 == ACTIVE_AE_DSP_ASYNC_ACTIVATE)
+        if (!IsActivated()) //< Prevent if already in activated state
```
Comment (AchimTuran) [inline xbmc/cores/AudioEngine/DSPAddons/ActiveAEDSP.cpp]: Why not moving the IsActivated() method call to Activate()?

File: xbmc/cores/AudioEngine/DSPAddons/ActiveAEDSP.cpp [hunk 2]
```diff
@@ -93,16 +92,19 @@ void CActiveAEDSP::OnApplicationMessage(KODI::MESSAGING::ThreadMessage* pMsg)
...
+            Activate(true);
+          }
+          else
+          {
+            Activate();
+          }
         }
       }
-      else if(pMsg->param1 == ACTIVE_AE_DSP_STATE_OFF)
+      else if(pMsg->param1 == ACTIVE_AE_DSP_STATE_OFF && IsActivated()) //< Prevent if already in deactive state with IsActivated check
```
Comment (AchimTuran) [inline xbmc/cores/AudioEngine/DSPAddons/ActiveAEDSP.cpp]: We also can move this call to Deactivate().

Comment (AlwinEsch) [inline xbmc/cores/AudioEngine/DSPAddons/ActiveAEDSP.cpp]: Yes is better on Activate, I'm was a bit fast and has only put to the source of the faults.

**Comment by AlwinEsch** [comment]
Is updated and everything works fine on start of Kodi now, about my thing with PVR (nothing there) there was one check missing and added for ADSP in second commit.

Also all loops are improved with a ++i.

From my side ok to merge.

jenkins build this please.

**Pull request merged**